Assets/Mirage/ObjectPolling/ObjectPoolManager.cs

```diff
@@ -0,0 +1,204 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Mirage.Runtime
+{
+    public class ObjectPoolingManager
+    {
+        #region Fields
+
+        private readonly Dictionary<int, object> _recycledObjects = new Dictionary<int, object>();
+        private ServerObjectManager _serverObjectManager;
+        private GameObject _parent;
+        internal readonly Dictionary<Guid, NetworkIdentity> _objectsAssetIds = new Dictionary<Guid, NetworkIdentity>();
+
+
+        #endregion
+
+        public ObjectPoolingManager(ServerObjectManager serverObjectManager, GameObject parent)
+        {
+            _serverObjectManager = serverObjectManager;
+            _parent = parent;
+        }
+
+        #region Custom Spawn Handlers
+
+        /// <summary>
+        ///     Spawn handler for all network spawning of things.
+        /// </summary>
+        /// <param name="msg">The message incoming from spawn handler.</param>
+        /// <returns>Return back the network identity of the object needing soawning.</returns>
+        internal NetworkIdentity SpawnObject(SpawnMessage msg)
+        {
+            NetworkIdentity spawnedObject = NetworkSpawnPool(msg.position, Quaternion.identity, msg.assetId, 1);
+
+            // Active the pooled object.
+            spawnedObject.gameObject.SetActive(true);
+
+            // return the game object back to mirage to have it finish spawning.
+            return spawnedObject;
+        }
+
+        /// <summary>
+        ///     Spawn handler for all network unspawning of things.
+        /// </summary>
+        /// <param name="spawned">What network identity we are going to unspawn and return back to network pool.</param>
+        internal void UnSpawnObject(NetworkIdentity spawned)
+        {
+            ObjectPooling<NetworkIdentity> p = default;
+
+            foreach (object pool in _recycledObjects.Values)
```

Comment:
why can't we use a dictionary look up here?

Reply:
What do you mean?